onnx/defs/schema.cc

```diff
@@ -191,6 +190,14 @@ void OpSchema::Verify(const NodeProto& node) const {
       fail_check("Unrecognized attribute: ", name, " for operator ", node.op_type());
     }
 
+    if (attr_proto.has_ref_attr_name()) {
+      if (!attr_proto.has_type() || attr_proto.type() != expected_type) {
```

Comment:
looks to me this check is also applicable when the attr doesn't have ref_attr_name (however you will need to guard the has_type check by ir_version, IIRC type was added in ir_version 2

Reply:
Actually if it is a solid attr_proto if the check will be covered under checker.cc L199-208.. This validation logic is aimed to be only conducted on the referencing attr_proto.